include/control_filters/exponential_filter.hpp

```diff
@@ -0,0 +1,133 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONTROL_FILTERS__EXPONENTIAL_FILTER_HPP_
+#define CONTROL_FILTERS__EXPONENTIAL_FILTER_HPP_
+
+#include <memory>
+#include <limits>
+#include <string>
+
+#include "filters/filter_base.hpp"
+
+#include "control_toolbox/filters.hpp"
+#include "exponential_filter_parameters.hpp"
+
+namespace control_filters
+{
+
+/***************************************************/
+/*! \class ExponentialFilter
+  \brief A exponential filter class.
+
+  This class implements a low-pass filter for
+  various data types.
+
+  \section Usage
+
+  The ExponentialFilter class is meant to be instantiated as a filter in
+  a controller but can also be used elsewhere.
+  For manual instantiation, you should first call configure()
+  (in non-realtime) and then call update() at every update step.
+
+*/
+/***************************************************/
+
+template <typename T>
+class ExponentialFilter : public filters::FilterBase<T>
+{
+public:
+  /*!
+   * \brief Configure the ExponentialFilter (access and process params).
+   */
+  bool configure() override;
+
+  /*!
+   * \brief Applies one iteration of the exponential filter.
+   *
+   * \param data_in input to the filter
+   * \param data_out filtered output
+   *
+   * \returns false if filter is not configured, true otherwise
+   */
+  bool update(const T & data_in, T & data_out) override;
+
+private:
+  rclcpp::Clock::SharedPtr clock_;
+  std::shared_ptr<rclcpp::Logger> logger_;
+  std::shared_ptr<exponential_filter::ParamListener> parameter_handler_;
+  exponential_filter::Params parameters_;
+  T last_smoothed_value;
+};
+
+template <typename T>
+bool ExponentialFilter<T>::configure()
+{
+  logger_.reset(
+    new rclcpp::Logger(this->logging_interface_->get_logger().get_child(this->filter_name_)));
+
+  // Initialize the parameters once
+  if (!parameter_handler_)
+  {
+    try
+    {
+      parameter_handler_ =
+        std::make_shared<exponential_filter::ParamListener>(this->params_interface_,
+                                                         this->param_prefix_);
+    }
+    catch (rclcpp::exceptions::ParameterUninitializedException & ex) {
+      RCLCPP_ERROR((*logger_), "Exponential filter cannot be configured: %s", ex.what());
+      parameter_handler_.reset();
+      return false;
+    }
+    catch (rclcpp::exceptions::InvalidParameterValueException & ex)  {
+      RCLCPP_ERROR((*logger_), "Exponential filter cannot be configured: %s", ex.what());
+      parameter_handler_.reset();
+      return false;
+    }
```

Comment:
```suggestion
    catch (const std::exception & ex) {
      RCLCPP_ERROR((*logger_), "Exponential filter cannot be configured: %s (type : %s)", ex.what(), typeid(e).name());
      parameter_handler_.reset();
      return false;
    }
```

Reply:
Similar to https://github.com/ros-controls/ros2_control/blob/53656e4636ba0472a5ee20d3a29f096ff50f6f5d/controller_manager/src/controller_manager.cpp#L544-L551

Reply:
Should I add a `catch (...)` as well?

Reply:
I move that also to a different issue, because this pattern is the same for every filter here.

Reply:
Sure. I agree. THis can be a different issue. May be we can add catch(..) as well.